core/capabilities/compute/compute.go

```diff
@@ -97,35 +105,96 @@ func generateID(binary []byte) string {
 	return fmt.Sprintf("%x", id)
 }
 
-func copyRequest(req capabilities.CapabilityRequest) capabilities.CapabilityRequest {
-	return capabilities.CapabilityRequest{
-		Metadata: req.Metadata,
-		Inputs:   req.Inputs.CopyMap(),
-		Config:   req.Config.CopyMap(),
+func (c *Compute) Execute(ctx context.Context, request capabilities.CapabilityRequest) (capabilities.CapabilityResponse, error) {
+	ch, err := c.enqueueRequest(ctx, request)
```

Comment:
Do we need to worry about the capability timing out in the engine if this queue gets too long?

Reply:
We do -- this is why I added the step-level timeout; we'll wait for a maximum of 2 minutes (which is incredibly generous) before interrupting a step (and 10 minutes for the whole workflow).

Reply:
Sorry, I'll be more precise, I'm not convinced the timer in the engine should start until it's running.  Users shouldn't be penalized if other tasks can block them.  I'm worried that I can DoS the compute capability by making N compute steps that have infinite loops and intentionally time out.

For now, maybe this is ok, we can re-evaluate before we open compute for general use...

Reply:
> I'm worried that I can DoS the compute capability by making N compute steps that have infinite loops and intentionally time out.

This shouldn't be possible because we apply a lower-level timeout to the individual WASM call; the default setting for this is 2s. I set the step-level timeout to be very large partly as an attempt to compensate for this.

What solution would you propose? We could ignore the engine timeout I suppose, but that feels dangerous IMO.